src/main/java/com/thevoxelbox/voxelsniper/brush/type/TreeSnipeBrush.java

```diff
@@ -43,18 +43,21 @@ public void handleCommand(String[] parameters, Snipe snipe) {
     @Override
     public void handleArrowAction(Snipe snipe) {
         BlockVector3 targetBlock = getTargetBlock().add(0, getYOffset(), 0);
-        single(targetBlock);
+        single(snipe, targetBlock);
     }
 
     @Override
     public void handleGunpowderAction(Snipe snipe) {
-        single(getTargetBlock());
+        single(snipe, getTargetBlock());
     }
 
-    private void single(BlockVector3 targetBlock) {
+    private void single(Snipe snipe, BlockVector3 targetBlock) {
         BlockState currentBlockData = getBlock(targetBlock.getX(), targetBlock.getY() - 1, targetBlock.getZ());
         setBlockType(targetBlock.getX(), targetBlock.getY() - 1, targetBlock.getZ(), BlockTypes.GRASS_BLOCK);
-        generateTree(this.treeType, targetBlock);
+        if (!generateTree(this.treeType, targetBlock)) {
```

Comment:
(Wrong spot, but) maybe leave a note placing trees thru this brush is currently not undo-able, because spigot changed linked method?

Reply:
Certainly! _Wrong spot too, u should also add a warning in FAWE then 🤔_

Reply:
Good point, https://github.com/IntellectualSites/FastAsyncWorldEdit/commit/69067fe8c8a87ec1c5a1528f605724dc71d74e43